.github/workflows/test-distros.yml

```diff
@@ -0,0 +1,60 @@
+name: Python Test on Linux distros
+
+# This workflow should only run when preparing releases to make sure linux distros other than
+# ubuntu work as expected.
+# In pull requests only run if this file has been changed
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/test-distros.yml'
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: ${{ matrix.container }}
+    strategy:
+      fail-fast: false
+      matrix:
+        container:
+          - "archlinux:latest"
+          - "debian:stable"
+          - "debian:testing"
+          - "fedora:latest"
+          - "fedora:rawhide"
+          - "opensuse/tumbleweed:latest"
+          - "ubuntu:devel"
```

Comment:
Pardon my ignorance but where do all these images come from and are they "pinned" somehow? Also, what do they have access to? The volume of code potential attack surface of an entire distribution is orders of magnitude bigger than for a simple Github Action (#802)

(not like I remember where they came from in the previous method either, apologies)

Reply:
The pull logic is the same as running the images locally, they are fetched from [docker hub](https://hub.docker.com/).

The reason these aren't pinned is to actually validate the running heads, hence the reason it should be considered to be run manually when needed.

Reply:
Thanks, so maybe we could have weekly runs with pinned images and manual runs without pinning?

But... is pinning less secure here? I mean, isn't pinning missing security updates? Aren't we supposed to upgrade operating systems all the time?

Funny how every dependency in #802 looks like a nail to pin but not here. Maybe there are differences and good reasons not to pin here but the corresponding rationales and policies should be summarized and referenced somewhere and I haven't seen this anywhere so far.

Reply:
This, IMHO, is outside the scope of this PR. I suggest we take that separately since this PR contains a good change that is worth getting in ASAP.